libgit/browser_file.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"io"
+
+	"github.com/pkg/errors"
+	billy "gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+type browserFile struct {
+	f *object.File
+	r io.ReadCloser
+}
+
+var _ billy.File = (*browserFile)(nil)
+
+func newBrowserFile(f *object.File) (*browserFile, error) {
+	r, err := f.Reader()
+	if err != nil {
+		return nil, err
+	}
+	return &browserFile{
+		f: f,
+		r: r,
+	}, nil
+}
+
+func (bf *browserFile) Name() string {
+	return bf.Name()
+}
+
+func (bf *browserFile) Write(_ []byte) (n int, err error) {
+	return 0, errors.New("Browser files can't be written")
+}
+
+func (bf *browserFile) Read(p []byte) (n int, err error) {
+	return bf.r.Read(p)
+}
+
+func (bf *browserFile) ReadAt(p []byte, off int64) (n int, err error) {
+	// Sadly go-git doesn't expose a `ReadAt` interface for this, but
+	// we can probably implement it if needed.  Instead, use a new
+	// Reader object and just scan starting from the beginning.
+	fullData := make([]byte, int64(len(p))+off)
+	r, err := bf.f.Reader()
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		_ = r.Close()
+	}()
+	n, err = r.Read(fullData)
```

Comment:
I'm a bit worried about reading large objects here. Instead of read everything into ram, perhaps we can have an upper count for the buffer here, and call `r.Read` multiple times until we reach the specified offset? Although on the other hand, files in git are usually small enough to fit in RAM so maybe it's fine this way for now.

Same for `Seek()` below, but I guess for that we should just aim for the real implementation.

Reply:
Good idea, done.